### PR TITLE
Java corrections

### PR DESCRIPTION
**Pull request by pablo-parra**
This PR tries to solve issues: #1 , #3 , #7 , #9 , #10 and #12

### CODE REVIEW

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java [hunk 1]
```diff
@@ -26,7 +25,7 @@
 import io.oasp.application.mtsj.usermanagement.dataaccess.api.UserEntity;
 
 @Entity
-@javax.persistence.Table(name = "Booking")
+@Table(name = "Booking")
 public class BookingEntity extends ApplicationPersistenceEntity implements Booking {
 
   @NotNull
```
Comment (maybeec) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java]: replace with @Column(nullable=false) on the getter.

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java [hunk 2]
```diff
@@ -58,7 +57,7 @@
 
   private UserEntity user;
 
-  @LazyCollection(LazyCollectionOption.FALSE)
+  @Basic(fetch = FetchType.LAZY)
```
Comment (maybeec) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java]: We sometimes have annotations on the field itself and somestimes on the getters.
For now we simply specified all annotations on the getters. So we should do it also here consistently.
In this special case you could even integrate the fetch specification into the existing oneToMany annotation.

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/dishmanagement/dataaccess/api/DishEntity.java [hunk 3]
```diff
@@ -37,10 +35,10 @@
 
   private ImageEntity image;
 
-  @LazyCollection(LazyCollectionOption.FALSE)
+  @Basic(fetch = FetchType.LAZY)
```
Comment (maybeec) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/dishmanagement/dataaccess/api/DishEntity.java]: join with existing annotations on the getter or simply move it

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/dishmanagement/dataaccess/api/DishEntity.java [hunk 4]
```diff
@@ -37,10 +35,10 @@
...
   private List<IngredientEntity> extras;
 
-  @LazyCollection(LazyCollectionOption.FALSE)
+  @Basic(fetch = FetchType.LAZY)
```
Comment (maybeec) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/dishmanagement/dataaccess/api/DishEntity.java]: join with existing annotations on the getter or simply move it

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/ordermanagement/dataaccess/api/OrderLineEntity.java [hunk 5]
```diff
@@ -32,7 +29,7 @@
 
   private DishEntity dish;
 
-  @LazyCollection(LazyCollectionOption.FALSE)
+  @Basic(fetch = FetchType.LAZY)
```
Comment (maybeec) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/ordermanagement/dataaccess/api/OrderLineEntity.java]: join with existing annotations on the getter or simply move it

File: java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java (hunk 1 above)
Comment (pablo-parra) [inline java/mtsj/core/src/main/java/io/oasp/application/mtsj/bookingmanagement/dataaccess/api/BookingEntity.java]: Maybe we can address this particular change in the validations PR #28

**Pull request merged**